src/matrix.h

```diff
@@ -0,0 +1,46 @@
+#ifndef MATRIX_H
+#define MATRIX_H
+
+typedef unsigned int matrix_size; 
+typedef double matrix_real_number;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Converts a 2d coordinate in a matrix to an index in a 2d array
+*/
+matrix_size coordToIndex(const matrix_size row, const matrix_size col, const matrix_size size);
+matrix_size isDiagonal(const matrix_size index, const matrix_size size);
+
+/**
+ * Multiplies 2 matrices together using this equation : output = firstMatrix x secondMatrix
+ * @param firstMatrix first matrix to multiply of size m * n
+ * @param secondMatrix second matrix to multiply of size n * p
+ * @param size array of size 3 containing m, n and p
+ * @param output matrix resulting of the matrix multiplication of the given matrices
+ * @param transposeFirstMatrix whether to transpose the first matrix when multiplying the matrices
+ */
+void matrixMultiply(const matrix_real_number* firstMatrix, const matrix_real_number* secondMatrix, const matrix_size size[3], matrix_real_number* output, const matrix_size transposeFirstMatrix);
```

Comment:
I just saw that the documentation of the functions are in the .h file. We always put them in the .c file so I would move that to the .c file. Also, there are certain functions without a definition.